### PR TITLE
Vignette: minimisation vs block and simple randomization

### LINKED ISSUE #33: Vignette: minimisation vs block and simple randomization
We plan to create a detailed vignette for the "Unbiased" R package, showcasing the implementation and effectiveness of Pocock's minimisation algorithm in comparison with simple and block randomization methods. This vignette will use a simulation approach, modeled after the footcell study, to demonstrate how these randomization methods impact the balance of covariates in a clinical trial setting.

Key aspects of the vignette include:

1. Data Simulation: Generating test data reflecting a clinical trial scenario, with covariates and treatment assignments based on Pocock's minimisation, simple randomization, and block randomization.
2. Monte Carlo Simulation: Conducting Monte Carlo simulations to assess the statistical properties and covariate balance achieved by each randomization method.
3. Statistical Analysis and Visualization:
    1. Employing the Standardised Mean Difference (SMD) test to compare covariate balances.
    1. Presenting selected examples from the simulations in a visual format, such as balance of sex across different randomization methods, to provide an intuitive understanding of the results.

Additional Enhancements:

- Implement an upper limit in the truncated distribution, such as capping age at 100.
- Replace boxplots with violin plots for a more detailed visualization of all covariates.
- Utilize a 'check balance' function for each randomization type to assess the balancing of covariates.
- Present chi-squared test results for each method, focusing on p-values.
- Explore the impact of different weighting schemes (e.g., 3:1) on key vs. other covariates.
- Include block randomization in the analysis and summary sections.

### Acceptance Criteria

1. Rich Content and Analysis: The vignette should thoroughly cover simulation, statistical analysis, and visual comparisons, including both SMD statistics and example plots from simulations.
2. Technical Accuracy and Clarity: Ensure the vignette is accurate, clear, and comprehensible, with well-documented R code for each step.
3. Visualizations: Include violin plots and other relevant visual aids to enhance understanding.
4. Integration with pkgdown: Compile and integrate the vignette into our existing pkgdown site for easy access and reference.

### PR DESCRIPTION
**Pull request by kamilsi**
This closes #33

### CODE REVIEW

File: vignettes/simulations.Rmd [hunk 1]
```diff
@@ -0,0 +1,851 @@
...
+        K = n_groups,
+        ratio = ratio,
+        groups = arm_names,
+        filledBlock = FALSE
+      )
+      getRandList(genSeq(rand))[1,]
+    })
+    df_list = tibble::tibble()
+    for (i in seq_len(strata_n)) {
+      local_df <- strata_grid %>%
```
Comment (lwalejko) [inline vignettes/simulations.Rmd:550]: Could you use `|>` pipe?

File: vignettes/simulations.Rmd [hunk 2]
```diff
@@ -0,0 +1,851 @@
...
+data.frame(hba1c, tpo2, age, wound_size) %>%
+  rename('wound size' = wound_size) %>%
+  pivot_longer(cols = everything(),
+               names_to = "parametr",
+               values_to = "n") %>%
+  mutate(
+    n = ceiling(n),
+    percent = round(n / 46, 2),
+    strata = c('<=9', '<=50', '<=55', '<=2')
+  ) %>%
```
Comment (lwalejko) [inline vignettes/simulations.Rmd:185]: Could you use `|>` pipe?

File: vignettes/simulations.Rmd [hunk 3]
```diff
@@ -0,0 +1,851 @@
...
+```
+
+Lists of randomization codes were generated for each iteration based on the function, assuming a block vector of c(3,6,9). This ensures blinding and facilitates ease in predicting which group the next patient would be assigned to.
+
+```{r, rand-data}
+# generate randomization lists for each iterations
+rand_data <- tibble::tibble()
+
+for (i in unique(data$simnr)) {
+  simulation_result <-
```
Comment (lwalejko) [inline vignettes/simulations.Rmd:568]: you can make this a function instead and generate randomization table on demand to conserve memory

**Review by lwalejko — CHANGES_REQUESTED** [review]

File: vignettes/simulations.Rmd [hunk 4]
```diff
@@ -0,0 +1,860 @@
...
+
+# create beta distribution parameters
+build_prior_df <- function(no_of_iterations, n) {
+  binary_matrix <- tribble(
+    ~ var,
+    ~ distribution,
+    ~ alpha, #1
+    ~ beta, #0
+    "sex", # Parameters provided by researchers
+    "beta",
```
Comment (kamilsi) [inline vignettes/simulations.Rmd:205]: I think you loose the `row tibble` advantage when you break the rows like this :-)

File: vignettes/simulations.Rmd [hunk 5]
```diff
@@ -0,0 +1,860 @@
...
+    SIMPLIFY = FALSE)
+  names(params_binary) <- binary_matrix$var
+  
+  params <-
+   params_binary |>
+    bind_rows() 
+  
+  return(params)
+}
+```
```
Comment (kamilsi) [inline vignettes/simulations.Rmd:254]: Let's skip this stem entirely and do simple Monte Carlo

File: vignettes/simulations.Rmd [hunk 6]
```diff
@@ -0,0 +1,860 @@
...
+        simstudy::defData(varname = "hba1c",
+                          formula = hba1c,
+                          dist = "binary") |>
+        simstudy::defData(varname = "tpo2",
+                          formula = tpo2,
+                          dist = "binary") |>
+        simstudy::defData(
+          varname = "age",
+          # correlation with diabetes type - assumptions - younger patients are more likely to have type I diabetes
+          formula = "(diabetes_type==0) * (-3.5)",
```
Comment (kamilsi) [inline vignettes/simulations.Rmd:289]: you can skip comparison and do something like `diabetes_type * 3.5` IMO

File: vignettes/simulations.Rmd [hunk 7]
```diff
@@ -0,0 +1,860 @@
...
+  as.data.frame() |>
+  rename(`power results [%]` = results_power) |>
+  gt()
+```
+
+## Conclusion
+
+Considering all three randomization methods: minimization, block randomization, and simple randomization, minimization performs the best in terms of covariate balance. Simple randomization has a significant drawback, as patient allocation to arms occurs randomly with equal probability. This leads to an imbalance in both the number of patients and covariate balance, which is also random.
+
+On the other hand, block randomization performs very well in balancing the number of patients in groups in a specified allocation ratio. However, its effect size power is lower for covariate balance compared to minimization.
```
Comment (kamilsi) [inline vignettes/simulations.Rmd:841]: What do you mean "effect size power"?

**Review by kamilsi — COMMENTED** [review]
Great work on this document, Ola! You've done a fantastic job in laying out the details of the simulation. I've included a few suggestions to further refine and clarify some points. Your effort in putting this together really shows, and these minor adjustments are just to enhance what is already a solid piece of work. Keep up the great effort!

File: vignettes/simulations.Rmd [hunk 8]
```diff
@@ -0,0 +1,860 @@
...
+    result <-
+      simulate_long |>
+      group_by(Simulation) |>
+      transmute(Simulation, n = sum(Value <= number)) |>
+      distinct()
+    
+    mean <- mean(result$n)
+    
+    return(mean)
+}
```
[the remark this answers is not in the document]
Comment (aleksandraduda2) [reply vignettes/simulations.Rmd:149]: right suggestion

**Review by JagGlo — APPROVED** [review]
Great work, Ola! This chunk of text you prepared is really impressive!
Plus: Vignette has now became an article.

**Review by lwalejko — APPROVED** [review]

**Pull request merged**